# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: startupcache/StartupCacheChild.h
# Commit: 6450088b6b73
# Full Hash: 6450088b6b73ffc17c79cd7097b1bfe30d00e207
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-09-14 21:29:41
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1656261 - Back out all recent StartupCache work r=RyanVM
#   
#   This backs out all work from bug 1627075 as well as all of its
#   descendents. There were a few conflicts when backing this out but
#   overall it was pretty clean, so I would say it's a fairly mild
# ==============================================================================

diff -r b0fabcd60430 -r 6450088b6b73 startupcache/StartupCacheChild.h
--- a/startupcache/StartupCacheChild.h	Mon Sep 14 17:32:40 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,40 +0,0 @@
-/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 2; -*- */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef StartupCacheChild_h
-#define StartupCacheChild_h
-
-#include "mozilla/scache/StartupCache.h"
-#include "mozilla/scache/PStartupCacheChild.h"
-#include "mozilla/scache/PStartupCacheParent.h"
-
-namespace mozilla {
-namespace ipc {
-class FileDescriptor;
-}
-
-namespace scache {
-
-using mozilla::ipc::FileDescriptor;
-
-class StartupCacheChild final : public PStartupCacheChild {
-  friend class mozilla::scache::StartupCache;
-  friend class mozilla::scache::StartupCacheListener;
-
- public:
-  StartupCacheChild() = default;
-
-  void Init();
-
- protected:
-  virtual void ActorDestroy(ActorDestroyReason aWhy) override;
-
-  void SendEntriesAndFinalize(StartupCache::Table& entries);
-};
-
-}  // namespace scache
-}  // namespace mozilla
-
-#endif  // StartupCacheChild_h